### DISCUSSION TITLE
Dynamic manifest.json generation based on external parameters

### DISCUSSION THREAD

**Discussion by SebasScript** [vercel/next.js#61130] [App Router]
I can dynamically create the manifest.json file like this
[Generate a Manifest file](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/manifest#generate-a-manifest-file)

is it possible to inject any information into the manifest.ts metadata route?  I would like to change name and icon of the manifest based on the user that creates the request.
I can access the cookies but they seem to be empty and there doesn't seem to be any other props that can be accessed.

**Comment by SebasScript** [comment]
I got a working solution now in which I host the manifest.json files on a cloud storage and load them dynamically by using the generateMetadata() function in my root layout.tsx.
Still curious though if this is also possible using manifest.ts

**Comment by rnnyrk** [comment]
Also looking for a solution to dynamically generate a manifest for different sites in a monorepo

**Comment by comxd** [comment]
In my case I want to generate a webmanifest dynamically based on query params.
Given that  the `manifest.ts` don't receive the `request` as argument of the exported `manifest()` function, I have forgot to use this special route handler fiel convention and created my custom route handler something like that:

```ts
// file: app/manifest.webmanifest/route.ts

import { NextRequest, NextResponse } from "next/server";

export const GET = (request: NextRequest) => {
  const searchParams = request.nextUrl.searchParams;
  const tenant = searchParams.get("t");

  // your validation and logic...

  const manifest = {
    // your manifest file content
  };

  return NextResponse.json(manifest);
}
```

PS: don't forgot to handle cache correctly.

**Comment by SebasScript** [comment]
Cool, thanks for the update, i will try this as soon as possible,
i couldn't find a solution, and worked on a prototype that creates the manifest dynamically in an s3 bucket based  and than in the 
root layout.tsx component uses the

```
generateMetadata() {
metadata: gets3BucketUrl()
}
```

function to get the url of the manifest based on page parameter similar to your solution. Not having to create a file on s3 is great!!
